src/test/kotlin/org/rust/ide/inspections/RsDeprecationInspectionTest.kt

```diff
@@ -0,0 +1,225 @@
+        trait Bar {
+            #[deprecated]
+            fn foo(&self);
+        }
+
+        struct Foo;
+
+        impl Bar for Foo {
+            fn foo(&self) {
+                println("foo")
+            }
+        }
+
+        fn main() {
+            let foo = Foo;
+            foo.<warning descr="'foo' is marked as deprecated">foo</warning>();
+        }
+    """)
+
+    fun `test deprecated enum`() = checkByText("""
+        #[deprecated]
+        enum Numbers {
+            One,
+            Two,
+            Three
+        }
+
+        fn main() {
+            <warning descr="'Numbers' is marked as deprecated">Numbers</warning>::One;
+        }
+    """)
+
+    fun `test deprecated enum item`() = checkByText("""
+        enum Numbers {
+            #[deprecated]
+            One,
+            Two,
+            Three
+        }
+
+        fn main() {
+            Numbers::<warning descr="'One' is marked as deprecated">One</warning>;
+        }
+    """)
+
+    fun `test deprecated type alias`() = checkByText("""
+        #[deprecated]
+        type Name = String;
+
+        fn main() {
+            let name: <warning descr="'Name' is marked as deprecated">Name</warning> = "test".to_string();
+        }
+    """)
+
+    fun `test deprecated variable`() = checkByText("""
+        #[deprecated]
+        static SOME_INT: i32 = 5;
+
+        fn main() {
+            let a = <warning descr="'SOME_INT' is marked as deprecated">SOME_INT</warning>;
+        }
+    """)
+
+    fun `test deprecated inline module`() = checkByText("""
+        #[deprecated]
+        mod foo {
+            pub static SOME_INT: i32 = 5;
+        }
+
+        fn main() {
+            let a = <warning descr="'foo' is marked as deprecated">foo</warning>::SOME_INT;
+        }
+    """)
+
+    fun `test deprecated variable in struct`() = checkByText("""
+        struct Foo {
+            #[deprecated]
+            pub x: i32
+        }
+
+        fn main() {
+            let a = Foo { <warning descr="'x' is marked as deprecated">x</warning>: 123 };
+        }
+    """)
+
+    fun `test deprecated non-inline module`() = checkByFileTree("""
+        //- main.rs
+            #[deprecated]
+            mod foo;
+
+            fn main() {
+                let a = /*caret*/<warning descr="'foo' is marked as deprecated">foo</warning>::SOME_INT;
+            }
+        //- foo.rs
+            pub static SOME_INT: i32 = 5;
+    """)
+
+    fun `test allow deprecated function`() = checkByText("""
+        #[deprecated]
+        pub fn foo() {
+        }
+
+        #[allow(deprecated)]
+        fn main() {
+            foo();
+        }
+    """)
+
+    fun `test allow deprecated statement`() = checkByText("""
+        #[deprecated]
+        pub fn foo() {
+        }
+
+        fn main() {
+            #[allow(deprecated)]
+            foo();
+        }
+    """)
+
+    fun `test allow deprecated item in module`() = checkByText("""
```

Comment:
Let's add test when deprecation warnings suppressed in the whole file
In this case, you should add `#![allow(deprecated)]` at the beginning of file